azure-pipelines.yml

```diff
@@ -1,15 +1,37 @@
+variables:
+  CI_COMMON_VERSION: "1.0"
 
-jobs:
-  - job: build_docker_images
-    displayName: Build Docker Images
-    condition: not(eq(variables['Build.SourceBranch'], 'refs/heads/master'))
+stages:
+- stage: build_common_docker_images
+  displayName: Build Common Docker Images
+  condition: always()
+  jobs:
+  - job:
+    condition: eq(variables['Build.SourceBranch'], 'refs/heads/master')
+    timeoutInMinutes: 360
```

Comment:
Unless there's a good reason to use a lower limit, perhaps just set this to 0 so it uses the maximum (5 hours)?

Reply:
The default is one hour as far as know? (I got a few timeout without this...)

Reply:
Right, this is a bit unintuitive, but:

no timeout specified - 1 hour
timeoutInMinutes: 0 - 5 hours
timeoutInMinutes: 360 - 6 hours (perhaps this rounds down to 5 hours as that's the max though?)

So yeah, if you explicitly set the timeoutInMinutes to 0, it should just use the highest limit instead of 1 hour (as I understand it).

Reply:
Hehe I didn't notice that, fun!